### PR TITLE
[QA-322] Adding the README.md, help function and IPC messages to reach the minimum viable product.

### PR DESCRIPTION
**Pull request by Aleksandras-V**
…imum viable product

### CODE REVIEW

**Review by RomainChaumontet — CHANGES_REQUESTED** [review]
When you launch ```./ipc_sendfile``` without any argument you don't have an error (you have one when you do the same with ipc_receivefile.

I tried with a file of 42k and it worked. But with a file of more than 1Mo the program crashed.
image


You can also modify your makefile (erase export lines, and just put qcc in the line 2)

**Review by RomainChaumontet — APPROVED** [review]
Great Work!

Code is clear, programs are running correctly.

File: ipc_receivefile.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+
+int ipc_receive_message(char* fileName)
+{
+
+	name_attach_t *att;
+	int rcvid;
+	msg_buf_t msg;
+	int status;
+	int messageStatus=1;
```
Comment (farnoodmah) [inline ipc_receivefile.c]: maybe int messageStatus = 1; can make the code more readable

File: ipc_receivefile.c [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+				{
+					perror("MsgError");
+					exit(EXIT_FAILURE);
+				}
+			}
+			else
+			{
+
+				iov_t siov[(msg.header.data_size/iov_block_size)+1];
+				int last_part_size = msg.header.data_size%iov_block_size;
```
Comment (farnoodmah) [inline ipc_receivefile.c]: Some comments above this can explain more about what is going on

File: ipc_sendfile.c [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    unsigned char* buffer=readFile(fileName); //getting the data into the buffer
+
+	SETIOV (&siov[0], &header, sizeof header); //setting header IOV
+
+	for(i=1;i<=fileSize/iov_block_size+1;i++){ //setting data IOVs
+		if (i == fileSize/iov_block_size+1){ //in case it is the last IOV
+			SETIOV (&siov[i], &buffer[(i-1)*iov_block_size], last_part_size);
+		} else { //else
+			SETIOV (&siov[i], &buffer[(i-1)*iov_block_size], iov_block_size);
+		}
```
Comment (farnoodmah) [inline ipc_sendfile.c:132]: some comments explaining the processes can be helpful too.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+DEBUG = -g
+CC = qcc
+LD = qcc
+
+TARGET = -Vgcc_ntox86_64
+
+CFLAGS += $(DEBUG) $(TARGET) -Wall
+LDFLAGS+= $(DEBUG) $(TARGET)
+
```
Comment (KinhTruongUnikie) [inline Makefile:9]: When I do make command on the terminal, it runs the code under clean. One suggestion is to move the "all" section above "clean"

Comment (Aleksandras-V) [reply Makefile:9]: Changed !

File: ipc_sendfile.c (hunk 3 above)
Comment (Aleksandras-V) [reply ipc_sendfile.c:132]: Done!

File: ipc_receivefile.c (hunk 2 above)
Comment (Aleksandras-V) [reply ipc_receivefile.c]: Done!

File: ipc_receivefile.c (hunk 1 above)
Comment (Aleksandras-V) [reply ipc_receivefile.c]: Done

**Review by farnoodmah — APPROVED** [review]
Great

**Review by KinhTruongUnikie — APPROVED** [review]

**Pull request merged**